lib/Sema/CSSimplify.cpp

```diff
@@ -7744,11 +7774,25 @@ void ConstraintSystem::addExplicitConversionConstraint(
   SmallVector<Constraint *, 3> constraints;
 
   auto locatorPtr = getConstraintLocator(locator);
-
+  auto coerceLocator = [&]() {
+    if (allowFixes && shouldAttemptFixes()) {
+      if (auto *expr = dyn_cast_or_null<CoerceExpr>(locator.getAnchor())) {
+        auto toTypeRepr = expr->getCastTypeLoc().getTypeRepr();
+        // Only adding this path for explicty coercions e.g _ = a as Int
+        // and also only for left-side is a DeclRefExpr or a
+        // explicit/implicit coercion e.g. Double(1) as Double
+        if (!expr->isImplicit() &&
+            !isa<ImplicitlyUnwrappedOptionalTypeRepr>(toTypeRepr) &&
+            (isa<DeclRefExpr>(expr->getSubExpr()) || isa<CoerceExpr>(expr->getSubExpr())))
+          return getConstraintLocator(expr, LocatorPathElt::ExplicitTypeCoercion());
+      }
+    }
+    return locatorPtr;
+  }();
```

Comment:
What is the reason for use of closures like this?

Reply:
Maybe it actually makes more sense to always add `explicit type coercion` bit to the locator and move check for sub-expression to the `attempt` method?

Reply:
Makes sense, moved to there :)